xarray/core/groupby.py

```diff
@@ -120,20 +134,25 @@ def __init__(self, obj, group, squeeze=False, grouper=None):
                              'dimension')
         full_index = None
 
-        if grouper is not None:
-            # time-series resampling
+        if grouper is not None and bins is not None:
+            raise ValueError("Can't specify both `grouper` and `bins`.")
+        elif grouper is not None or bins is not None:
             index = safe_cast_to_index(group)
             if not index.is_monotonic:
                 # TODO: sort instead of raising an error
                 raise ValueError('index must be monotonic for resampling')
             s = pd.Series(np.arange(index.size), index)
-            first_items = s.groupby(grouper).first()
+            if grouper is not None:
+                # probably time-series resampling
+                first_items = s.groupby(grouper).first()
+            else:
+                first_items = s.groupby(pd.cut(index, bins)).first()
             if first_items.isnull().any():
                 full_index = first_items.index
                 first_items = first_items.dropna()
-            bins = first_items.values.astype(np.int64)
-            group_indices = ([slice(i, j) for i, j in zip(bins[:-1], bins[1:])] +
-                             [slice(bins[-1], None)])
+            sbins = first_items.values.astype(np.int64)
+            group_indices = ([slice(i, j) for i, j in zip(sbins[:-1], sbins[1:])] +
```

Comment:
The nature of `group_indices` here is why we need a monotonic group -- otherwise we need to use `unique_value_groups` to pick out indices, and I couldn't figure out a way to use pandas to get that information from a `TimeGrouper` (this is possibly a bug, see https://github.com/pydata/pandas/issues/13152).

But I don't think you need to use this code path here. Instead try inserting something like the following just above where we set `full_index = None`:
```python
if bins is None:
    group = xr.DataArray(pd.cut(group.values, bins), group.coords, name=group.name)
```

Then I think this can go through the normal code path, which involves calling `unique_value_groups`, anyways.